File: corda/java/contracts/src/main/java/com/helloBlock/contracts/helloBlockContract.java
package com.helloBlock.contracts;

import net.corda.core.contracts.CommandData;
import net.corda.core.contracts.CommandWithParties;
import net.corda.core.contracts.Contract;
import net.corda.core.contracts.TypeOnlyCommandData;
import net.corda.core.transactions.LedgerTransaction;
import com.helloBlock.states.helloBlockState;
import org.jetbrains.annotations.NotNull;

import static net.corda.core.contracts.ContractsDSL.requireSingleCommand;
import static net.corda.core.contracts.ContractsDSL.requireThat;

// ************
// * Contract *
// ************
// Contract and state.
public class helloBlockContract implements Contract {
    // Used to identify our contract when building a transaction.
    public static final String ID = "com.helloBlock.contracts.helloBlockContract";

    // Contract code.
    @Override
    public void verify(@NotNull LedgerTransaction tx) throws IllegalArgumentException {
        final CommandWithParties<Commands.Send> command = requireSingleCommand(tx.getCommands(), Commands.Send.class);
        requireThat(require -> {
        final helloBlockState helloBlock = tx.outputsOfType(helloBlockState.class).get(0);
        require.using("Sender must sign the message.", helloBlock.getOrigin().getOwningKey().equals(command.getSigners().get(0)));
        return null;
        });
    }

    // Used to indicate the transaction's intent.
    public interface Commands extends CommandData {
        class Send extends TypeOnlyCommandData {}
    }
}

File: corda/kotlin/contracts/src/main/kotlin/com/helloBlock/contracts/helloBlockContract.kt
package com.helloBlock.contracts

import net.corda.core.contracts.*
import net.corda.core.contracts.Requirements.using
import net.corda.core.transactions.LedgerTransaction
import com.helloBlock.states.helloBlockState

// ************
// * Contract *
// ************
// Contract and state.
class helloBlockContract: Contract {
    companion object {
        // Used to identify our contract when building a transaction.
        const val ID = "com.helloBlock.contracts.helloBlockContract"
    }

    // Contract code.
    override fun verify(tx: LedgerTransaction) {
        val command = tx.commands.requireSingleCommand<Commands.Send>()
        val helloBlock = tx.outputsOfType<helloBlockState>().single()
        "Sender must sign the message." using (helloBlock.origin.owningKey == command.signers.single())
    }

    // Used to indicate the transaction's intent.
    interface Commands : CommandData {
        class Send : Commands
    }
}